providers/implementations/keymgmt/dh_kmgmt.c

```diff
@@ -222,6 +266,33 @@ static int dh_get_key_params(void *key, OSSL_PARAM params[])
     return dh_get_dpk_params(key, params);
 }
 
+static int dh_iskey(const void *key)
+{
+    return DH_get0_pub_key(key) != NULL;
+}
+
+static int dh_cmpkey(const void *key1, const void *key2)
+{
+    if (BN_cmp(DH_get0_pub_key(key1), DH_get0_pub_key(key2)) != 0)
```

Comment:
hmm this is an interesting check... Are we only interested in the public key comparison here?

Reply:
I basically copied the code from dh_ameth.c. Interestingly enough, the legacy function pointer is called `pub_cmp`, so it's a bit more explicit. I was thinking that it would be up to the provider to decide exactly how detailed the comparison should be. In this implementation, I chose to do like we did before. For some implementations where the private half is hidden (typically HSMs), this would be the only possible comparison. 

With the assumption that the private half actually corresponds to the public part, this comparison is enough. I suppose that key validation will ensure this validity?